addons/website_mail/controllers/main.py

```diff
@@ -47,111 +46,90 @@ def _message_post_helper(res_model='', res_id=None, message='', token='', token_
-            res = res.sudo()
+            record = record.sudo()
         else:
             raise NotFound()
-    return res.with_context({'mail_create_nosubscribe': nosubscribe}).message_post(body=message,
-                                                                                   message_type=kw.pop('message_type', False) or "comment",
-                                                                                   subtype=kw.pop('subtype', False) or "mt_comment",
+    return record.with_context(mail_create_nosubscribe=nosubscribe).message_post(body=message,
+                                                                                   message_type=kw.pop('message_type', "comment"),
+                                                                                   subtype=kw.pop('subtype', "mt_comment"),
                                                                                    author_id=author_id,
                                                                                    **kw)
 
+
 class WebsiteMail(http.Controller):
 
     @http.route(['/website_mail/follow'], type='json', auth="public", website=True)
     def website_message_subscribe(self, id=0, object=None, message_is_follower="on", email=False, **post):
         # TDE FIXME: check this method with new followers
-        cr, uid, context = request.cr, request.uid, request.context
-
-        partner_obj = request.registry['res.partner']
-        user_obj = request.registry['res.users']
-
-        _id = int(id)
-        _message_is_follower = message_is_follower == 'on'
-        _object = request.registry[object]
+        res_id = int(id)
+        is_follower = message_is_follower == 'on'
+        record = request.env[object].browse(res_id)
 
         # search partner_id
-        public_id = request.website.user_id.id
-        if uid != public_id:
-            partner_ids = [user_obj.browse(cr, uid, uid, context).partner_id.id]
+        if request.env.user != request.website.user_id:
+            partner_ids = request.env.user.partner_id.ids
         else:
             # mail_thread method
-            partner_ids = _object._find_partner_from_emails(
-                cr, SUPERUSER_ID, [_id], [email], context=context, check_followers=True)
+            partner_ids = record.sudo()._find_partner_from_emails([email], check_followers=True)
             if not partner_ids or not partner_ids[0]:
                 name = email.split('@')[0]
-                partner_ids = [partner_obj.create(cr, SUPERUSER_ID, {'name': name, 'email': email}, context=context)]
-
+                partner_ids = request.env['res.partner'].sudo().create({'name': name, 'email': email}).ids
         # add or remove follower
-        if _message_is_follower:
-            _object.check_access_rule(cr, uid, [_id], 'read', context)
-            _object.message_unsubscribe(cr, SUPERUSER_ID, [_id], partner_ids, context=context)
+        if is_follower:
```

Comment:
I comment those lines, and remove the `sudo()`before `message_subscribe` : if i'm on a blog, and i want to subscribe to the blog, by introducing my email adress, this controller returns a `Access Denied by ACLs for operation: write, uid: 4, model: blog.blog`
So, we are gonna keep those checks ;)